### PR TITLE
 Update parser binary filename to include OS and architecture

### PR DESCRIPTION
**Pull request by kpanuragh**
Modified the binary filename to include the operating system and architecture details. This change ensures that the correct binary is downloaded
 for the user's specific environment, improving compatibility and reducing potential errors during execution.

### CODE REVIEW

File: src/support/parser.ts [hunk 1]
```diff
@@ -37,7 +37,9 @@ export const setParserBinaryPath = (context: vscode.ExtensionContext) => {
 
 const downloadBinary = async (context: vscode.ExtensionContext) => {
     const binaryVersion = "0.1.6";
-    const filename = `php-parser-${binaryVersion}`;
+    const osPlatform = os.platform();
+    const osArch = os.arch();
```
Comment (eznix86) [inline src/support/parser.ts:41]: we need to ensure that the binary return has x64 and arm64 as `os.arch()` returns those possible values.

 'arm' , 'arm64' , 'ia32' , 'loong64' , 'mips' , 'mipsel' , 'ppc' , 'ppc64' , 'riscv64' , 's390' , 's390x' , and 'x64' 
which is different from https://github.com/SamuelNitsche/vs-code-php-parser-cli/releases/tag/0.1.9 

@joetannenbaum

Comment (joetannenbaum) [reply src/support/parser.ts:41]: Yup! Thank you. Working on integrating this now.

Comment (romansh) [reply src/support/parser.ts:41]: Is this related to the 'php-parser-0.1.6: Exec format error' preventing the extension from running?

Comment (kpanuragh) [reply src/support/parser.ts:41]: Yes

**Comment by joetannenbaum** [comment]
Thank you for the PR! Closing this in favor of a different solution inspired by this.

**Pull request closed without merging**